sanitize/sanitize.go

```diff
@@ -84,22 +98,30 @@ func (s *Sanitizer) sanitizeField(field reflect.Value) {
 		t := field.Type()
 		entityName := t.Name()
 
-		entitySkipFields, hasEntitySkips := entityLevelExemptedFields[entityName]
+		// specificEntityName is used to identify plugin or partial types
+		// it is useful for building exempted fields
+		specificEntityName, entitySchema, err := s.fetchEntitySchema(entityName, field)
+		if err != nil {
+			warningMessage := fmt.Sprintf("Error fetching schema for entity: %s %v\n"+
```

Comment:
in case of error, do we also want to inform user they they may want to manually redact any sensitive info in this entity?